src/univariate/continuous/beta.jl

```diff
@@ -29,18 +29,22 @@ External links
 struct Beta{T<:Real} <: ContinuousUnivariateDistribution
     α::T
     β::T
+    Beta{T}(α::T, β::T) where {T} = new{T}(α, β)
+end
 
-    function Beta{T}(α::T, β::T) where T
-        @check_args(Beta, α > zero(α) && β > zero(β))
-        new{T}(α, β)
-    end
+function Beta(α::T, β::T) where {T<:Real}
+    @check_args(Beta, α > zero(α) && β > zero(β))
+    return Beta{T}(α, β)
+end
+
+function Beta(α::T, β::T, ::NoArgCheck) where {T<:Real}
+    return Beta{T}(α, β)
 end
 
-Beta(α::T, β::T) where {T<:Real} = Beta{T}(α, β)
 Beta(α::Real, β::Real) = Beta(promote(α, β)...)
 Beta(α::Integer, β::Integer) = Beta(Float64(α), Float64(β))
```

Comment:
I am not sure if using `float` instead of `Float64` is not better (the difference is minimal - only probably for `BigInt` argument in practical cases).

Reply:
good point indeed, these hardcoded Float64 never felt right

Reply:
I did this in the last commit